src/renderer/ducks/beacon/sagas.ts

```diff
@@ -0,0 +1,44 @@
+import {all, call, put, takeEvery, PutEffect, CallEffect} from "redux-saga/effects";
+import {startLocalBeacon, removeBeacon, addBeacon} from "./actions";
+import {endDockerImagePull, startDockerImagePull} from "../network/actions";
+import {getNetworkConfig} from "../../services/eth2/networks";
+import {BeaconChain} from "../../services/docker/chain";
+import {SupportedNetworks} from "../../services/eth2/supportedNetworks";
+import database from "../../services/db/api/database";
+
+function* startLocalBeaconSaga({
+    payload: {network, ports},
+}: ReturnType<typeof startLocalBeacon>): Generator<PutEffect | CallEffect, void, BeaconChain> {
+    // Pull image first
+    yield put(startDockerImagePull());
```

Comment:
do we handle if this errors? Like image doesn't exists or internet dc. 

It's annoying we don't have any progress (on slow network this could go on forever).

Reply:
some parts of this issue will be handled with #301  ?

Reply:
yeah true